gooddata-server-oauth2-autoconfigure/src/main/kotlin/BearerTokenReactiveAuthenticationManagerResolver.kt

```diff
@@ -63,17 +69,38 @@ private class PersistentApiTokenAuthenticationManager(
             .cast(BearerTokenAuthenticationToken::class.java)
             .flatMap { authToken ->
                 getOrganizationFromContext().flatMap { organization ->
-                    client.getUserByApiToken(organization.id, authToken.token).map { user ->
-                        UserContextAuthenticationToken(organization, user).also {
-                            logger.logFinishedAuthentication(
-                                organization.id,
-                                user.id,
-                                "API Token",
-                            ) {
-                                withTokenId(user.usedTokenId)
+                    client.getUserByApiToken(organization.id, authToken.token)
+                        .map { user ->
+                            UserContextAuthenticationToken(organization, user).also {
+                                logger.logFinishedAuthentication(
+                                    organization.id,
+                                    user.id,
+                                    "API Token",
+                                ) {
+                                    withTokenId(user.usedTokenId)
+                                }
+
+                                auditClient.recordLoginSuccess(
+                                    orgId = organization.id,
+                                    userId = user.id,
+                                    source = sourceIp,
+                                    sessionContextType = AuthMethod.API_TOKEN,
+                                    sessionContextIdentifier = user.usedTokenId
+                                ).subscribe()
                             }
```

Comment:
I believe we should avoid initializing a nested flow inside an existing one (`.subscribe()`)
Instead we should incomporate the audit log call into the existing reactive pipeline.
With sth as this:
```suggestion
                      .flatMap { user ->
                            val token = UserContextAuthenticationToken(organization, user)
                            logger.logFinishedAuthentication(
                                organization.id,
                                user.id,
                                "API Token",
                            ) {
                                withTokenId(user.usedTokenId)
                            }

                            auditClient.recordLoginSuccess(
                                orgId = organization.id,
                                userId = user.id,
                                source = sourceIp,
                                sessionContextType = AuthMethod.API_TOKEN,
                                sessionContextIdentifier = user.usedTokenId
                            ).thenReturn(token)
                        }
```